platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainQueryService.java

```diff
@@ -172,27 +218,30 @@ private void warnIfAutoProvisionedToolchainUsedWithoutRepositoryDefinitions(Inst
         }
     }
 
-    private JavaToolchain downloadToolchain(JavaToolchainSpec spec) {
+    private JavaToolchain downloadToolchain(JavaToolchainSpec spec, Set<JavaInstallationCapability> requiredCapabilities) {
```

Comment:
No, the `spec` is end-user configuration, whereas these capabilities are only used internally. It would be interesting to consider exposing them as public API in the future (#29278 related).